searchcore/src/vespa/searchcore/proton/flushengine/flushengine.cpp

```diff
@@ -274,10 +286,79 @@ FlushEngine::prune()
             EventLogger::flushPrune(createName(*handler, oldestFlushed.second), oldestFlushed.first);
         }
         handler->flushDone(oldestFlushed.first);
+        prune_done(strategy_ids_for_finished_flushes, kv.second);
     }
+    prune_flushing_strategies(strategy_ids_for_finished_flushes);
     return true;
 }
 
+void
+FlushEngine::prune_done(std::vector<uint32_t>& strategy_ids_for_finished_flushes, const std::vector<PruneMeta>& prune_metas)
+{
+    for (auto& prune_meta : prune_metas) {
+        _flush_history->prune_done(prune_meta._flush_id);
+        strategy_ids_for_finished_flushes.emplace_back(prune_meta._strategy_id);
+    }
+}
+
+void
+FlushEngine::prune_flushing_strategies(std::vector<uint32_t> strategy_ids_for_finished_flushes)
```

Comment:
Consider passing by const ref or `std::move`-ing at the callsites, since these seem to currently pass by value